src/make_trade.ts

```diff
@@ -31,7 +31,7 @@ const TRADE_TIMEOUT_SECONDS = 300;
 const { concat, hexlify, hexValue } = ethers.utils;
 
 export async function makeTrade(
-  tokenListUrl: string,
+  tokenListUrl: string | null,
```

Comment:
Is this really null or undefined?
```suggestion
  tokenListUrl: string | undefined,
```

Reply:
**edit**: Tried it out, and indeed it is `undefined` when un-set and not `null`:

```ts
task("foobar", "Foo Bar")
  .addOptionalParam("param", "Some param")
  .setAction(async ({ param }) => console.log(param));
```

Will print out `undefined` when no parameter is specified.